src/extractor/restriction_parser.cpp

```diff
@@ -244,14 +242,15 @@ bool RestrictionParser::ShouldIgnoreRestriction(const std::string &except_tag_st
 
     // Be warned, this is quadratic work here, but we assume that
     // only a few exceptions are actually defined.
-    std::vector<std::string> exceptions;
-    boost::algorithm::split_regex(exceptions, except_tag_string, boost::regex("[;][ ]*"));
+    const std::regex delimiter_re("[;][ ]*");
```

Comment:
Similar problem to the Travis Xenial [upgrade attempt](https://github.com/Project-OSRM/osrm-backend/pull/5901#discussion_r640162103).
Boost.Regex and GCC5 appears to be buggy with respect to `[abi:cxx11]` tags, so using this as an opportunity to replace it with `std::regex`.

Reply:
Nice!